### DISCUSSION TITLE
Importance of rabbit_disk_monitor

### DISCUSSION THREAD

**Discussion by Vishnu-0809** [rabbitmq/rabbitmq-server#11271] [Questions]
Hi, 
I was going through the docs about rabbitmq disk alarm [disk space alarms](https://www.rabbitmq.com/docs/disk-alarms#overview) , in which it is mentioned that when the available disk space below the configured value, a log message is generated and also the producers are blocked.

I have rabbitmq running in a container and I can see a process called rabbit_disk_monitor which I believe is responsible for disk alarms and above mentioned functionality, but when rabbit_disk_monitor process is killed I dont see it is restarted.
So for any reason if the rabbit_disk_monitor process is killed will the functionalities mentioned in above documentation are not performed or are they carried out by some other processes ?
I am trying to understand the significance of rabbit_disk_monitor and effect on application when it is killed

**Comment by lukebakken** [comment]
Please tell us the version of RabbitMQ and Erlang you are using, and on what operating system.

**Reply by Vishnu-0809** [reply]
RabbitMQ version - 3.13.0
Erlang version - 26.1.1
OS - Rocky Linux 8

**Comment by lukebakken** [comment]
`rabbit_disk_monitor` should never be killed in normal operation. What do your logs say as to the reason that it is being killed?
